src/topology/uniform_space/basic.lean

```diff
@@ -1043,16 +1046,32 @@ instance : has_bot (uniform_space α) :=
   is_open_uniformity :=
     assume s, by simp [is_open_fold, subset_def, id_rel] {contextual := tt } } ⟩
 
+instance : has_inf (uniform_space α) :=
+⟨λ u₁ u₂,
+  @uniform_space.replace_topology _
+    (u₁.to_topological_space ⊓ u₂.to_topological_space) (uniform_space.of_core
+    { uniformity  := u₁.uniformity ⊓ u₂.uniformity,
+      refl        := le_inf u₁.refl u₂.refl,
+      symm        := le_inf
+        (le_trans (map_mono inf_le_left) u₁.symm) (le_trans (map_mono inf_le_right) u₂.symm),
+      comp        := le_inf
+        (le_trans (lift'_mono inf_le_left le_rfl) u₁.comp)
+        (le_trans (lift'_mono inf_le_right le_rfl) u₂.comp) }) $
+    eq_of_nhds_eq_nhds $ λ a, begin
```

Comment:
I couldn't work out a nice way to prove this directly from the constructor, but I'm not at all familiar with the API here so I might have been missing something obvious.

Reply:
LGTM now. If the CI is happy, please merge it.

bors d+